pkg/client/run.go

```diff
@@ -67,44 +55,31 @@ func (c *SonobuoyClient) Run(cfg *RunConfig) error {
 			continue
 		}
 
-		obj := unstructured.Unstructured{}
-		if err := runtime.DecodeInto(scheme.Codecs.UniversalDecoder(), ext.Raw, &obj); err != nil {
+		obj := &unstructured.Unstructured{}
+		if err := runtime.DecodeInto(scheme.Codecs.UniversalDecoder(), ext.Raw, obj); err != nil {
 			return errors.Wrap(err, "couldn't decode template")
 		}
-
-		err := createObject(c.DynamicClientPool(), &obj, mapper)
+		name, err := c.dynamicClient.Name(obj)
 		if err != nil {
+			return errors.Wrap(err, "could not get object name")
+		}
+		namespace, err := c.dynamicClient.Namespace(obj)
+		if err != nil {
+			return errors.Wrap(err, "could not get object namespace")
+		}
+		newObj, err := c.dynamicClient.CreateObject(obj)
```

Comment:
It looks like you're ignoring this error....?

Reply:
the error gets handled on line 75

i agree though it's a weird pattern and may warrant a comment.